reference_guide/project_model/sdk.md

```diff
@@ -39,3 +39,68 @@ Sdk projectSdk = ProjectRootManager.getInstance(project).getProjectSdk();
   ```
   
 See the following [code sample](https://github.com/JetBrains/intellij-sdk-docs/blob/master/code_samples/project_model/src/com/intellij/tutorials/project/model/ProjectSdkAction.java) to get more familiar with SDK manipulation tool set.
+
+## Working with your own SDK
+
+To create your own SDK, You need to create a class extends `SdkType`, and set your recommended SDK name, SDK home path, a presentable name. <br/>
+Finally you need to implement `isValidSdkHome` to do the SDK validation, and this example just check if the given path is a directory.
+
+```java
+import com.intellij.openapi.projectRoots.*;
```

Comment:
The methods used in example either self-descriptive or contain javadoc. I'd say it's better to add documentation to `setupSdkPath` and provide a link to `SdkType` class.

Reply:
Link provided.